### ISSUE TITLE
File upload does not work anymore in v3

### ISSUE THREAD

**Issue by vkocjancic** [mbdavid/LiteDB#495]
Hi.

This type of Upload method call fails miserably, taking into account only id of image:
`db.FileStorage.Upload('1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59', '$/gallery/1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59', ms);`
where ms is memory stream populated with image.

I also added metadata:
```
var dict = new Dictionary<string, BsonValue>();
dict.Add("extension", ".jpg");
db.FileStorage.SetMetadata('$/gallery/1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59', new BsonDocument(dict));
```

It produces this result in DB:
`{"_id":"1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59","filename":"1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59","mimeType":"application/octet-stream","length":{"$numberLong":"2480"},"chunks":1,"uploadDate":{"$date":"2017-02-20T12:46:35.1980000Z"},"metadata":{}}`

Basically, filename is ignored and metadata is not set. It worked in v2.0.4.

**Comment by vkocjancic** [comment]
@mbdavid... To be fair, I think that v2.0.4 had a bug, where path was eventually stored in id and id in path (at least the data shows that). But even if I turn that around with v3.0.1, binary data is still not saved and meta data not set.

**Comment by mbdavid** [comment]
Hi @vkocjancic, I made an unit test to check and the problem as you told: to use SetMetadata you must pass FileId, not FileName. There is no build-in index in Filename field. Take a look on unit test that I made and checks if is your case.

**Comment by vkocjancic** [comment]
@mbdavid grooby. Metadata now works. 

There is still a problem, though.
Calling this:
`db.FileStorage.Upload('1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59', '$/gallery/1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59', ms);`
Produces:
`{"_id":"1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59","filename":"1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59","mimeType":"application/octet-stream","length":{"$numberLong":"2480"},"chunks":1,"uploadDate":{"$date":"2017-02-20T12:46:35.1980000Z"},"metadata":{}}`

Two problems persist: 

1. Id is '1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59' even though it was  '$/gallery/1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59' in v2.0.4
1. Binary data is not stored. At all.

Calling same lines in v2.0.4 worked without a glitch.

**Comment by mbdavid** [comment]
Hi @vkocjancic, there are some syntax change from v2. In v2, if you do not pass ID but pass filename, LiteDB assumes filename as ID. In v3 ID is mandatory.

In v3 there is a new upload operation, based in Stream (now you can open a file to write). Is your "ms" at begin file (position 0)?

I update my unit test and are working fine. Can you fork LiteDB and write your test to reproduce this problem?

**Comment by vkocjancic** [comment]
@mbdavid Sorry, I don't have that much time to play about with forks and unit tests. :(
` ms.Seek(0, SeekOrigin.Begin); ` is called prior Upload method.

But why is filename not stored in full? In v2.0.4, filename was stored in id!? and filename was the same as in v3.0.1 (thus, id was ` $/gallery/1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59 `. Should I specify ` $/gallery/1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59 ` as id and not filename in v3.0.1?

**Comment by vkocjancic** [comment]
@mbdavid Sorry, I don't have that much time to play about with forks and unit tests. :(
` ms.Seek(0, SeekOrigin.Begin); ` is called prior Upload method.

But why is filename not stored in full? In v2.0.4, filename was stored in id!? and filename was the same as in v3.0.1 (thus, id was ` $/gallery/1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59 `. Should I specify ` $/gallery/1020d6eb-e5fb-4c94-8a21-c0ea2f3a4b59 ` as id and not filename in v3.0.1?
